tezos/new_context/src/working_tree/serializer.rs

```diff
@@ -378,13 +517,94 @@ pub fn deserialize(data: &[u8], storage: &mut Storage) -> Result<Entry, Deserial
             let message = message.to_vec();
 
             Ok(Entry::Commit(Box::new(Commit {
-                parent_commit_hash: HashId::new(parent_commit_hash),
-                root_hash: HashId::new(root_hash).ok_or(MissingRootHash)?,
+                parent_commit_hash: parent_commit_hash,
+                root_hash: root_hash.ok_or(MissingRootHash)?,
                 time,
                 author: String::from_utf8(author)?,
                 message: String::from_utf8(message)?,
             })))
         }
+        ID_INODE_POINTERS => {
+            let inode = deserialize_inode_tree(&data[1..], storage, store)?;
+            let inode_id = storage.add_inode(inode)?;
+
+            Ok(Entry::Tree(inode_id.into()))
+        }
+        _ => Err(UnknownID),
+    }
+}
+
+fn deserialize_inode_tree(
+    data: &[u8],
+    storage: &mut Storage,
+    store: &ContextKeyValueStore,
+) -> Result<Inode, DeserializationError> {
+    use DeserializationError::*;
+
+    let mut pos = 0;
+
+    let depth = data.get(pos..pos + 4).ok_or(UnexpectedEOF)?;
+    let depth = u32::from_ne_bytes(depth.try_into()?);
+
+    let nchildren = data.get(pos + 4..pos + 8).ok_or(UnexpectedEOF)?;
+    let nchildren = u32::from_ne_bytes(nchildren.try_into()?);
+
+    let npointers = data.get(pos + 8..pos + 8 + 1).ok_or(UnexpectedEOF)?;
+    let npointers = u8::from_ne_bytes(npointers.try_into()?);
+
+    pos += 9;
+
+    let data_length = data.len();
+    let mut pointers: [Option<PointerToInode>; 32] = Default::default();
+
+    while pos < data_length {
+        let index = data.get(pos..pos + 1).ok_or(UnexpectedEOF)?;
+        let index = u8::from_ne_bytes(index.try_into()?);
+
+        pos += 1;
+
+        let bytes = data.get(pos..).ok_or(UnexpectedEOF)?;
+        let (hash_id, nbytes) = deserialize_hash_id(bytes)?;
+
+        pos += nbytes;
+
+        let inode_id = match store.get_value(hash_id.ok_or(MissingHash)?) {
+            Ok(data) => deserialize_inode(data.unwrap().as_ref(), storage, store)?,
+            Err(_) => panic!(),
```

Comment:
There is a panic with no information here. Is a panic the only meaningful action here? what situations could cause this?

Reply:
Same as the previous comment, it is now handled correctly.